components/script/dom/document.rs

```diff
@@ -1234,11 +1234,37 @@ impl Document {
         }
     }
 
+    /// Return the title if one is specified
+    fn get_title(&self) -> Option<DOMString> {
+        let title = self.GetDocumentElement().and_then(|root| {
+            if root.namespace() == &ns!(svg) && root.local_name() == &local_name!("svg") {
+                // Step 1.
+                root.upcast::<Node>()
+                    .child_elements()
+                    .find(|node| {
+                        node.namespace() == &ns!(svg) && node.local_name() == &local_name!("title")
+                    })
+                    .map(DomRoot::upcast::<Node>)
+            } else {
+                // Step 2.
+                root.upcast::<Node>()
+                    .traverse_preorder(ShadowIncluding::No)
+                    .find(|node| node.is::<HTMLTitleElement>())
+            }
+        });
+
+        title.map(|title| {
+            // Steps 3-4.
```

Comment:
Without the spec link, these steps don't refer to anything any longer. Please add the specification link.